tools/sigh.ts

```diff
@@ -957,21 +975,54 @@ function uploadCodeHealthStats(request, data: string[], testResult: boolean) {
   keepProcessAlive = true; // Tell the runner to not exit.
 }
 
-function spawnTool(toolPath: string, args: string[]) {
-  return saneSpawn('node', [...nodeFlags, '--no-warnings', toolPath, ...args]);
+// Single place to put all common node flags for running node tools.
+function prepNodeToolSpawn(toolPath: string, args: string[]) : string[] {
+  return [...nodeFlags, '--no-warnings', toolPath, ...args];
+}
+
+function spawnNodeToolSync(toolPath: string, args: string[]) : boolean {
+  return saneSpawnSync('node', prepNodeToolSpawn(toolPath, args));
 }
 
-function devServer(args: string[]) {
+function spawnNodeTool(toolPath: string, args: string[]) : ChildProcess {
+  return saneSpawn('node', prepNodeToolSpawn(toolPath, args));
+}
+
+// Single place to put all optional dependencies for the devServer.
+function prepDevServerOptionalDeps() {
   getOptionalDependencies(['chokidar'], 'The devServer command');
-  return spawnTool('build/tools/dev_server/dev-server.js', args);
+}
+
+const devServerPath = 'build/tools/dev_server/dev-server.js';
+
+function devServer(args: string[]) : boolean {
+  prepDevServerOptionalDeps();
+  return spawnNodeToolSync(devServerPath, args);
+}
+
+let devServerProcess: ChildProcess = null;
```

Comment:
Looks like you can move this inside the devServerAsync function

Reply:
I wasn't sure, given that it happens later, but I guess it's a closure so it's ok.